server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TestTaskPlugin.java

```diff
@@ -469,4 +492,70 @@ protected UnblockTestTasksRequestBuilder(ElasticsearchClient client,
         }
     }
 
+    private static class OriginAssertingInterceptor implements TransportInterceptor {
```

Comment:
This thing is fairly tricky but it looked like the path of least resistance as far as asserting that the origin is properly set by the task management API. It doesn't really prove the interaction with security, but I have the reindex-with-security REST tests for that.